zcash_client_sqlite/src/wallet/init/migrations/v_transactions_net.rs

```diff
@@ -0,0 +1,586 @@
+        WalletDb,
+    };
+
+    #[test]
+    fn v_transactions_net() {
+        let data_file = NamedTempFile::new().unwrap();
+        let mut db_data = WalletDb::for_path(data_file.path(), tests::network()).unwrap();
+        init_wallet_db_internal(&mut db_data, None, &[add_transaction_views::MIGRATION_ID])
+            .unwrap();
+
+        // Create two accounts in the wallet.
+        let usk0 =
+            UnifiedSpendingKey::from_seed(&tests::network(), &[0u8; 32][..], AccountId::from(0))
+                .unwrap();
+        let ufvk0 = usk0.to_unified_full_viewing_key();
+        db_data
+            .conn
+            .execute(
+                "INSERT INTO accounts (account, ufvk) VALUES (0, ?)",
+                params![ufvk0.encode(&tests::network())],
+            )
+            .unwrap();
+
+        let usk1 =
+            UnifiedSpendingKey::from_seed(&tests::network(), &[1u8; 32][..], AccountId::from(1))
+                .unwrap();
+        let ufvk1 = usk1.to_unified_full_viewing_key();
+        db_data
+            .conn
+            .execute(
+                "INSERT INTO accounts (account, ufvk) VALUES (1, ?)",
+                params![ufvk1.encode(&tests::network())],
+            )
+            .unwrap();
+
+        // - Tx 0 contains two received notes of 2 and 5 zatoshis that are controlled by account 0.
+        db_data.conn.execute_batch(
+            "INSERT INTO blocks (height, hash, time, sapling_tree) VALUES (0, 0, 0, '');
+            INSERT INTO transactions (block, id_tx, txid) VALUES (0, 0, 'tx0');
+
+            INSERT INTO received_notes (tx, output_index, account, diversifier, value, rcm, nf, is_change)
+            VALUES (0, 0, 0, '', 2, '', 'nf_a', false);
+            INSERT INTO received_notes (tx, output_index, account, diversifier, value, rcm, nf, is_change)
+            VALUES (0, 3, 0, '', 5, '', 'nf_b', false);").unwrap();
+
+        // - Tx 1 creates two notes of 2 and 3 zatoshis for an external address, and a change note
+        //   of 2 zatoshis. This is representative of a historic transaction where no `sent_notes`
+        //   entry was created for the change value.
+        db_data.conn.execute_batch(
+            "INSERT INTO blocks (height, hash, time, sapling_tree) VALUES (1, 1, 1, '');
+            INSERT INTO transactions (block, id_tx, txid) VALUES (1, 1, 'tx1');
+            UPDATE received_notes SET spent = 1 WHERE tx = 0;
+            INSERT INTO sent_notes (tx, output_pool, output_index, from_account, to_account, to_address, value)
+            VALUES (1, 2, 0, 0, NULL, 'addra', 2);
+            INSERT INTO sent_notes (tx, output_pool, output_index, from_account, to_account, to_address, value, memo)
+            VALUES (1, 2, 1, 0, NULL, 'addrb', 3, X'61');
+            INSERT INTO received_notes (tx, output_index, account, diversifier, value, rcm, nf, is_change)
+            VALUES (1,    2, 0, '', 2, '', 'nf_c', true);").unwrap();
```

Comment:
Non-blocking:
```suggestion
            VALUES (1, 2, 0, '', 2, '', 'nf_c', true);").unwrap();
```

Reply:
This space was actually intended, so that the output indices would line up.